src/components/ProductTrainingContext/index.tsx

```diff
@@ -156,11 +165,10 @@ const useProductTrainingContext = (tooltipName: ProductTrainingTooltipName, shou
     useEffect(() => {
         if (shouldShow) {
             registerTooltip(tooltipName);
-            return () => {
-                unregisterTooltip(tooltipName);
-            };
         }
-        return () => {};
+        return () => {
+            unregisterTooltip(tooltipName);
+        };
```

Comment:
We should have un-registered the tooltip if `shouldShow` was `true`. This caused https://github.com/Expensify/App/issues/57569